engine/src/main/java/org/terasology/rendering/opengl/fbms/DisplayResolutionDependentFBOs.java

```diff
@@ -120,11 +123,15 @@ private void createFBOs() {
         }
     }
 
-    // TODO: Pairing FBOs for swapping functionality
     public void swapReadWriteBuffers() {
-        FBO fbo = get(READONLY_GBUFFER);
-        fboLookup.put(READONLY_GBUFFER, get(WRITEONLY_GBUFFER));
-        fboLookup.put(WRITEONLY_GBUFFER, fbo);
-        notifySubscribers();
+        buffersSwapped = !buffersSwapped;
```

Comment:
Not sure how late into the review you wrote this comment.

My PR description now answers this question, but just to reiterate : It's not just about swapping the variables, it's the side effects caused due to the swap that I'm concerned about (update calls to every subscriber).

This might get a bit long, but..
`     CURRENT                    NEW     
sda
`